infra/Pos/Diffusion/Subscription/Dns.hs

```diff
@@ -82,8 +84,12 @@ dnsSubscriptionWorker oq networkCfg DnsDomains{..} keepaliveTimer nextSlotDurati
         -- Try to subscribe to some peer.
         -- If they all fail, wait a while before trying again.
         subscribeToOne dnsPeersList
-        retryInterval >>= delay
-        subscribeAlts dnsPeersVar (index, alts)
+        interval <- retryInterval retryCounter
+        case interval of
+            Nothing -> logError "dnsSubscriptionWorker: DNS timeout failure"
```

Comment:
Here we give up forever? Should the program quit? Should we just have a maximum delay?